src/core/io/platform/f4xx/ADCf4xx.cpp

```diff
@@ -84,109 +98,189 @@ void ADCf4xx::initADC(uint8_t num_channels) {
     /** Configure the global features of the ADC (Clock, Resolution, Data
      * Alignment and number of conversion)
      */
-    halADC.Instance                   = ADC1;
-    halADC.Init.ClockPrescaler        = ADC_CLOCK_SYNC_PCLK_DIV2;
-    halADC.Init.Resolution            = ADC_RESOLUTION_12B;
-    halADC.Init.ScanConvMode          = ENABLE;
-    halADC.Init.ContinuousConvMode    = ENABLE;
-    halADC.Init.DiscontinuousConvMode = DISABLE;
-    halADC.Init.ExternalTrigConvEdge  = ADC_EXTERNALTRIGCONVEDGE_NONE;
-    halADC.Init.ExternalTrigConv      = ADC_SOFTWARE_START;
-    halADC.Init.DataAlign             = ADC_DATAALIGN_RIGHT;
-    halADC.Init.NbrOfConversion       = num_channels;
-    halADC.Init.DMAContinuousRequests = ENABLE;
-    halADC.Init.EOCSelection          = ADC_EOC_SEQ_CONV;
-
-    __HAL_RCC_ADC1_CLK_ENABLE();
-    HAL_ADC_Init(&halADC);
+    uint8_t adcNum         = getADCNum();
+    ADC_HandleTypeDef* adc = &halADC[adcNum];
+    // Set instance to the ADC peripheral being using
+    switch (adcNum) {
+    case ADC1_SLOT:
+        halADC[adcNum].Instance = ADC1;
+        break;
+    case ADC2_SLOT:
+        halADC[adcNum].Instance = ADC2;
+        break;
+    case ADC3_SLOT:
+        halADC[adcNum].Instance = ADC3;
+        break;
+    default:
+        return; // Should never get here
+    }
```

Comment:
This could be done when you initialize halADC since the Instance is just a pointer to the ADC memory address